src/DynamoCoreWpf/Interfaces/IWatchHandler.cs

```diff
@@ -24,16 +27,17 @@ namespace Dynamo.Interfaces
     public interface IWatchHandler
     {
         event Action<string> RequestSelectGeometry;
-        WatchViewModel Process(dynamic value, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback);
+        WatchViewModel Process(dynamic value, ObservableCollection<PortModel> outports, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback);
```

Comment:
Any reason the type signature of `outports` couldn't be `IEnumerable<string>`? This would decouple the code from `ObservableCollection` and `PortModel`. It would eliminate implementation details of `PortModel` from `IWatchHandler`, which makes the code a bit easier to think about. I personally would name this something like `preferredDictionaryOrdering`.